packages/core-db/test/app/ethereum/ethereum-block-processor.spec.ts

```diff
@@ -68,14 +68,38 @@ describe('Block Subscription', () => {
       .should.deep.equal([0, 1], `Incorrect blocks received!`)
   }).timeout(timeout)
 
-  it('processes old blocks starting at 1', async () => {
+  it('honors earliest block', async () => {
     blockProcessor = new EthereumBlockProcessor(db, 1)
     await blockProcessor.subscribe(provider, blockListener)
 
     const blocks: Block[] = await blockListener.waitForSyncToComplete()
 
-    blocks.length.should.equal(1)
-    blocks[0].number.should.equal(1)
+    blocks.length.should.equal(0)
+  }).timeout(timeout)
+
+  it('processes blocks starting at 1', async () => {
+    blockProcessor = new EthereumBlockProcessor(db, 1)
+    await blockProcessor.subscribe(provider, blockListener)
+
+    let blocks: Block[] = await blockListener.waitForSyncToComplete()
+    blocks.length.should.equal(0)
+
+    await tokenContract.transfer(
+      ownerWallet.address,
+      recipientWallet.address,
+      sendAmount * 2
+    )
+
+    blocks = await blockListener.waitForReceive(2)
+    blocks.length.should.equal(2, `Incorrect number of blocks received!`)
+
+    blocks
+      .map((x) => x.number)
+      .sort()
```

Comment:
Are the blocks in a non-deterministic pattern?

Reply:
Yep! We start syncing past blocks and current blocks at the same time, so order is non-deterministic until synced.

Reply:
(note: this is also one of the motivations for the downstream QueuedPersistedProcessor setup that enforces order on sequential data that may be received out of order.)